src/file-selector.ts

```diff
@@ -120,27 +120,32 @@ function flatten<T>(items: any[]): T[] {
     ], []);
 }
 
-function fromDataTransferItem(item: DataTransferItem, entry?: FileSystemEntry | null) {
+async function fromDataTransferItem(item: DataTransferItem, entry?: FileSystemEntry | null) {
     // Check if we're in a secure context; due to a bug in Chrome (as far as we know)
     // the browser crashes when calling this API (yet to be confirmed as a consistent behaviour).
     //
     // See:
     // - https://issues.chromium.org/issues/40186242
     // - https://github.com/react-dropzone/react-dropzone/issues/1397
     if (globalThis.isSecureContext && typeof (item as any).getAsFileSystemHandle === 'function') {
-        return (item as any).getAsFileSystemHandle()
-            .then(async (h: any) => {
-                const file = await h.getFile();
-                file.handle = h;
-                return toFileWithPath(file);
-            });
+        const h = await (item as any).getAsFileSystemHandle();
+        if (h === null) {
+            throw new Error(`${item} is not a File`);
+        }
+        // It seems that the handle can be `undefined` (see https://github.com/react-dropzone/file-selector/issues/120),
+        // so we check if it isn't; if it is, the code path continues to the next API (`getAsFile`).
+        if (h !== undefined) {
+            const file = await h.getFile();
+            file.handle = h;
+            return toFileWithPath(file);
+        }
```

Comment:
Wouldn't this be a safer fallback? Isn't it nice to try `getFile` if `getAsFileSystemHandle` does not return the expected data for some other unknown reason? 

```suggestion
        const h = await (item as any).getAsFileSystemHandle();
        if(!h) {
	        const file = await h.getFile();
	        if(!file) {
	        	throw new Error(`${item} is not a File`);
	        }
            file.handle = h;
            return toFileWithPath(file);
        }
```

And then you can skip the 
```ts
	const file = item.getAsFile(); 
    if (!file) {
        throw new Error(`${item} is not a File`);
    }
```

below, because it will always try `getFile`?

Reply:
```
if(!h) {
	        const file = await h.getFile();
```

That will essentially throw an exception because it will evaluate if `h` is `undefined` or `null`. We want the opposite of that. Note that `getFile` and `getAsFile` are 2 different APIs that exist on different objects.